mmds/src/data_store.rs

```diff
@@ -35,15 +35,43 @@ impl Mmds {
         self.is_initialized
     }
 
-    pub fn put_data(&mut self, data: Value) {
-        // TODO: we should add a data validator and only accept Strings, arrays & dictionaries
-        // https://github.com/firecracker-microvm/firecracker/issues/401
+    /// This method validates the data from a PATCH or PUT request and returns true
+    /// if the data only contains Strings, arrays and dictionaries
+    pub fn is_data_valid(&self, data: &Value) -> bool {
```

Comment:
I would return a `Result` here instead (but it's just a matter of personal preference):

```rust
pub fn validate_data(data: &Value) -> Result<(), Error> {
    if let Some(map) = data.as_object() {
        for key in map.keys() {
            Mmds::validate_data(&map[key])?;
        }
    } else if let Some(array) = data.as_array() {
        for obj in array {
            Mmds::validate_data(&obj)?;
        }
    } else {
        if !data.is_string() {
            return Err(Error::UnsupportedValueType);
        }
    }
    Ok(())
}
```

then propagate the error with the `?` operator.

Reply:
Also, there's no need to make this a member function. Nothing in the implementation needs `&self`.

Reply:
@aghecenco Thanks for the suggestions!